sbws/lib/relaylist.py

```diff
@@ -107,6 +131,48 @@ def master_key_ed25519(self):
             return None
         return key.rstrip('=')
 
+    @property
+    def consensus_valid_after(self):
+        network_status_document = self._from_ns('document')
+        if network_status_document:
+            return getattr(network_status_document, 'valid_after', None)
+        return None
+
+    @property
+    def last_consensus(self):
+        if len(self._consensus_timestamps) >= 1:
+            return self._consensus_timestamps[-1]
+        return None
+
+    def _set_new_consensus_timestamps(self, timestamp=None):
+        # It has been already updated or there's no timestamp yet
+        if self.consensus_valid_after is not None:
+            if self.last_consensus is not None:
+                if self.consensus_valid_after > self.last_consensus:
+                    self._consensus_timestamps.append(
+                        self.consensus_valid_after
+                        )
+            else:
+                self._consensus_timestamps.append(self.consensus_valid_after)
+        elif (self.last_consensus is not None
+              and timestamp > self.last_consensus):
+            self._consensus_timestamps.append(timestamp)
+        else:
+            self._consensus_timestamps.append(
+                datetime.utcnow().replace(microsecond=0))
+
+    @property
+    def _remove_old_consensus_timestamps(
```

Comment:
This doesn't seem like a property me; it looks a lot more like it should be a function.